File: matiasmasso/Mat.NET/DTO/DTOStaffJornada.cs
using System;

namespace DTO
{
    public class DTOStaffJornada : DTOBaseGuid
    {
        public DTOStaff Staff { get; set; }
        public DateTime FchFrom { get; set; }
        public DateTime FchTo { get; set; }
        public DTOStaffHoliday.Cods Cod { get; set; }
        public string Obs { get; set; }

        public DTOStaffJornada() : base()
        {
        }

        public DTOStaffJornada(Guid oGuid) : base(oGuid)
        {
        }

        public static DTOStaffJornada Factory(DTOStaff oStaff)
        {
            DTOStaffJornada retval = new DTOStaffJornada();
            {
                var withBlock = retval;
                withBlock.Staff = oStaff;
                withBlock.Cod = DTOStaffHoliday.Cods.Treball;
            }
            return retval;
        }
    }
}

File: matiasmasso/Mat.NET/DT2/DTOStaffJornada.vb
Public Class DTOStaffJornada
    Inherits DTOBaseGuid
    Property Staff As DTOStaff
    Property FchFrom As DateTime
    Property FchTo As DateTime
    Property Cod As DTOStaffHoliday.Cods
    Property Obs As String

    Public Sub New()
        MyBase.New
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
    End Sub

    Shared Function Factory(oStaff As DTOStaff) As DTOStaffJornada
        Dim retval As New DTOStaffJornada
        With retval
            .Staff = oStaff
            .Cod = DTOStaffHoliday.Cods.Treball
        End With
        Return retval
    End Function

End Class
